### PR TITLE
Add libsawtooth receipt store feature doc

### PR DESCRIPTION
**Pull request by isabeltomb**
Add a feature doc to the community > planning > feature section of
splinter.dev for the libsawtooth receipt store trait.

Signed-off-by: Isabel Tomb <tomb@bitwise.io>

### CODE REVIEW

File: community/planning/libsawtooth_receipt_store.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  - SQLite
+  - PostgreSQL
+
+## Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+### Proposed ReceiptStore Trait
+
+```
+/// Interface for performing CRUD operations on transaction receipts
```
Comment (agunde406) [inline community/planning/libsawtooth_receipt_store.md:63]: The formatting of this section is a little wonky.

File: community/planning/libsawtooth_receipt_store.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+Transaction receipts contain information related to the execution of a
+transaction. The receipt store allows for information related to the execution
+of a transaction that should not be stored in state to be saved to an underlying
+database.
+
+The `TransactionReceiptStore` is a struct that is a wrapper around an
+`OrderedStore`. The `OrderedStore` is a key value store indexed by a type. The
+receipt store uses transaction id as the key, transaction receipt as the value
+and is indexed by u64s. The `OrderedStore` trait is implemented for various
+backends including lmdb, redis and btree. The `OrderedStore` maintains the order
```
Comment (agunde406) [inline community/planning/libsawtooth_receipt_store.md:26]: Nitpick: Everywhere you say lmdb it should be LMDB (its done in some places) also id should be ID

**Review by agunde406 — CHANGES_REQUESTED** [review]

File: community/planning/libsawtooth_receipt_store.md (hunk 1 above)
Comment (isabeltomb) [reply community/planning/libsawtooth_receipt_store.md:63]: Fixed

File: community/planning/libsawtooth_receipt_store.md (hunk 2 above)
Comment (isabeltomb) [reply community/planning/libsawtooth_receipt_store.md]: Updated

**Review by agunde406 — APPROVED** [review]

**Review by peterschwarz — APPROVED** [review]

**Pull request merged**